### PR TITLE
Minor Android UI code cleanups

### PR DESCRIPTION
**Pull request by hackbar**
Minor cleanups, including making a proper GameCardView class (instead of an anonymous one), and creating a Platform enum.

### CODE REVIEW

**Comment by degasus** [comment]
@mahdihijazi What do you think about this PR and #6086 ?

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+/** View for showing a single game, meant for the TV interface. */
+public class GameCardView extends ImageCardView
+{
+	private static final int UNSELECTED_COLOR_ID = R.color.tv_card_unselected;
+	private final int UNSELECTED_COLOR;
+	private int selectedColor;
+
+	public GameCardView(Context context) {
+		super(context);
+		UNSELECTED_COLOR = getResources().getColor(UNSELECTED_COLOR_ID);
```
Comment (mahdihijazi) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java]: Avoid using this deprecated method and use ContextCompat.getColor() instead

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	protected void onRestoreInstanceState(Parcelable state)
+	{
+		SavedState savedState = (SavedState) state;
+		super.onRestoreInstanceState(savedState.getSuperState());
+		selectedColor = savedState.getSelectedColor();
+	}
+
+	public void setSelectedBackgroundColor(int selectedColorId)
+	{
+		this.selectedColor = getResources().getColor(selectedColorId);
```
Comment (mahdihijazi) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java]: same comment about getResources().getColor()

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+import android.content.Context;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.v17.leanback.widget.ImageCardView;
+
+import org.dolphinemu.dolphinemu.R;
+
+/** View for showing a single game, meant for the TV interface. */
+public class GameCardView extends ImageCardView
```
Comment (mahdihijazi) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java]: This class looks unneeded at all, I didn't test this but I think the old implementation was wrong and this PR followed the same path.

Why we don't use a simple xml based layout for the ImageCardView inside GameRowPresenter.onCreateViewHolder() and just use a simple color selector for the infoAreaBackgroundColor? do we really need to the custom class, shouldn't a simple color selector be able to switch the colors based on the selected state?

Another way could be to use a custom style as mentioned here:
https://developer.android.com/reference/android/support/v17/leanback/widget/ImageCardView.html

Looks like the custom style has the ability to change the infoBackgroundColor
https://chromium.googlesource.com/android_tools/+/882ee91503f28d1fd60ba939c5b37db6b9d4ad29/sdk/extras/android/support/v17/leanback/res/values/styles.xml#87

Comment (hackbar) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java]: That would be a lot better to use a selector and not having a custom class.

But I don't think any part of ImageCardView supports a ColorStateList though. infoAreaBackgroundColor is just a Color, even through styles, and I couldn't get View.setBackgroundTintMode to work with the ImageCardView.

Comment (mahdihijazi) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java]: I just tested setInfoAreaBackground() with drawable selector and it worked holder.cardParent.setInfoAreaBackground(ContextCompat.getDrawable(holder.cardParent.getContext(), R.drawable.game_cube_selector));

not sure about the View.setBackgroundTintMode() I don't see where you used this in the PR, or it something you wanted to use?

**Comment by mahdihijazi** [comment]
I noticed also that this PR adds new row to the main screen for "All Games" section, not sure if this was a bug in the old implementation and we should get it back @degasus?

my personal opinion is that adding this row doesn't add any value for the users, I would stick to the old UI.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java (hunk 3 above)
Comment (hackbar) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/GameCardView.java]: Ah, that makes sense. I didn't try to use drawable. I've updated to use that, please take a look.

Tint mode was just another thing I tried, but didn't work.

**Comment by hackbar** [comment]
Yep, this bring back the "All Games" row, it was hidden before due to a bug. I can remove it, as it would clean things up. I'm not a big fan of it.

**Comment by mahdihijazi** [comment]
@degasus this LGTM apart from the "All Games" row, if that is fine feel free to merge this PR.

**Comment by hackbar** [comment]
I added another commit to get rid of the 'All Games' section. Let me know what you think.

**Comment by hackbar** [comment]
I rebased this. Can you take a look at the last commit removing the 'all games' row?

**Comment by mahdihijazi** [comment]
Everything looks fine and I test it, "All Games" row is gone, thank you!

**Pull request merged**